src/meta/src/manager/background_deleter.rs

```diff
@@ -0,0 +1,105 @@
+// Copyright 2022 Singularity Data
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::sync::Arc;
+
+use itertools::Itertools;
+use risingwave_common::catalog::TableId;
+use tokio::sync::oneshot::Sender;
+use tokio::sync::{mpsc, oneshot};
+use tokio::task::JoinHandle;
+
+use super::SourceId;
+use crate::storage::MetaStore;
+use crate::stream::{GlobalStreamManagerRef, SourceManagerRef};
+use crate::MetaResult;
+
+pub type StreamingJobBackgroundDeleterRef = Arc<StreamingJobBackgroundDeleter>;
+
+#[derive(Debug)]
+// TODO(zehua): just use `TableId` instead of `StreamingJobId` after we remove source manager.
+pub enum StreamingJobId {
+    TableId(TableId),
+    SinkId(TableId),
+    SourceId(SourceId),
+}
+
+/// Used to delete actor, fragment, source and so on.
+/// When we drop a streaming job in a frontend, meta will drop it from meta store and notify
+/// frontends that object is dropped. The other things in compute node or storage related to the
+/// object will be drop in `StreamingJobBackgroundDeleter` in the background thread asynchronously.
+pub struct StreamingJobBackgroundDeleter(mpsc::UnboundedSender<Vec<StreamingJobId>>);
```

Comment:
What will happen if a new job is created while the delete task has not been executed yet? 🤔

Reply:
The new job will be created as usual. We have deleted the information in meta store, and it will get a different object id.